src/GafferImageUI/ImageView.cpp

```diff
@@ -393,12 +432,86 @@ bool ImageView::keyPress( const GafferUI::KeyEvent &event )
 				);
 				return true;
 			}
+			if(event.key == "Home")
+			{
+				
+				V2i viewport = viewportGadget()->getViewport();
+				V2i viewportCenter(viewport.x / 2, viewport.y / 2);
+				Box3f bound = m_imageGadget->bound();
+				V2i imageCenter((bound.max.x - bound.min.x) / 2, (bound.max.y - bound.min.y) / 2);
+				Box3f frame(
+					Imath::V3f(imageCenter.x - viewportCenter.x, imageCenter.y - viewportCenter.y, 0),
+					Imath::V3f(imageCenter.x + viewportCenter.x, imageCenter.y + viewportCenter.y, 0)
+				);
+				viewportGadget()->frame(frame);
+				return true;
+			}
+            if(event.key == "Plus")
+            {
+                float zoomLevel = computeZoomLevel();
+                float roundLevel;
+
+                if((zoomLevel + EPSILON) < 1.f)
+                {
+                    zoomLevel = std::min(100.f, 1 / zoomLevel);
+                    roundLevel = round_down_to_even(zoomLevel);
+                    
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * .5;
+                    }
+                    roundLevel = 1 / roundLevel;
+                }
+                else
+                {
+                    roundLevel = round_up_to_even(zoomLevel);
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * 2;
+                    }
+                }
+                zoomLevelPlug()->setValue(roundLevel) ;
+				
+                return true;
+
+            }
+            if(event.key == "Minus")
+            {
+                float zoomLevel = computeZoomLevel();
+                float roundLevel;
+                if((zoomLevel - EPSILON) < 1.f)
+                {
+                    zoomLevel = std::min(1000.0, round(1 / zoomLevel));
+                    roundLevel = round_up_to_even(zoomLevel);
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * 2;
+                    }
+                    roundLevel = 1 / roundLevel;
+                }
+                else
+                {
+                    roundLevel = round_down_to_even(round(zoomLevel));
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * .5;
```

Comment:
My intention is to mimic what Nuke is doing as a basis but I'm open to new ideas.

That would be:

- Zoom will be displayed as percentage but internally it's a float representing the zooming ratio.
- Hitting + or - adjusts the zoom by incrementing or decrementing the zoom factor ( current level 2, next level 3, implementation at the moment follow the original idea we discussed together at Image Engine).
- this kind of zooming will use snapping ( the reason for me is if we dont offer shortcut to specific zoom level like Nuke ( CTLR+3 => zoom factor 3, ALT+3 => zoom factor 1/3), that make sense to allow for it using the +/-.

Looking at Nuke in more details ( I never really paid so much attention to zooming in a viewport before doing that work!), I can see it's not a linear scale to zoom in/out, that make sense it would be too long to go through each level simply by incrementing.
The 1.41 factor is appealing and mimic the aperture scale on a camera.